frontend/src/app/services/search.service.ts

```diff
@@ -41,21 +41,19 @@ export class SearchService {
     public async search(queryModel: QueryModel
     ): Promise<SearchResults> {
         const user = await this.authService.getCurrentUserPromise();
-        const esQuery = queryModel.toEsQuery();
-        const query = new QueryDb(esQuery, queryModel.corpus.name, user.id);
-        query.started = new Date(Date.now());
-        const results = await this.elasticSearchService.search(
-            queryModel
-        );
-        query.total_results = results.total.value;
-        query.completed = new Date(Date.now());
-        this.queryService.save(query);
+        const request = () => this.elasticSearchService.search(queryModel);
 
-        return {
-            fields: queryModel.corpus.fields.filter((field) => field.resultsOverview),
-            total: results.total,
-            documents: results.documents,
-        } as SearchResults;
+        let resultsPromise: Promise<SearchResults>;
+
+        if (user.enableSearchHistory) {
+            resultsPromise = this.searchAndSave(queryModel, user, request);
+        } else {
+            resultsPromise = request();
+        }
```

Comment:
I'm not always a fan of ternary operator, but here it saves a lot of lines. Feel free to ignore.
```suggestion
        const resultPromise: Promise<SearchResults> = user.enableSearchHistory ? 
            this.searchAndSave(queryModel, user, request) : request();
```